Agent crash

**Describe the problem**

My agent seg faulted following some peer update.

**To Reproduce**

Not sure how, it's intermittent.

The stack log:

```console
5-01-17T21:25:40Z DEBG client/internal/routemanager/client.go:228: triggered route state update for Peer +s5VWGcyr6ZZiLHtFIrbJra9ndBc78q1djklt4gKPFA=, state: Connected
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x228 pc=0x9772d7]

goroutine 8547 [running]:
github.com/pion/ice/v3.(*Agent).ok(0x0)
        /home/runner/go/pkg/mod/github.com/netbirdio/ice/v3@v3.0.0-20240315174635-e72a50fcb64e/agent.go:176 +0x17
github.com/pion/ice/v3.(*Agent).Close(0x0)
        /home/runner/go/pkg/mod/github.com/netbirdio/ice/v3@v3.0.0-20240315174635-e72a50fcb64e/agent.go:928 +0x1c
github.com/netbirdio/netbird/client/internal/peer.(*WorkerICE).closeAgent(0xc00059b6c0, 0x5?)
        /home/runner/work/netbird/netbird/client/internal/peer/worker_ice.go:258 +0xa9
github.com/netbirdio/netbird/client/internal/peer.(*WorkerICE).reCreateAgent.func1(0x6)
        /home/runner/work/netbird/netbird/client/internal/peer/worker_ice.go:225 +0x16e
github.com/pion/ice/v3.(*Agent).onConnectionStateChange(0xc001910028?, 0xc000b21fb0?)
        /home/runner/go/pkg/mod/github.com/netbirdio/ice/v3@v3.0.0-20240315174635-e72a50fcb64e/agent_handlers.go:46 +0x5c
created by github.com/pion/ice/v3.(*Agent).connectionStateRoutine in goroutine 6083
        /home/runner/go/pkg/mod/github.com/netbirdio/ice/v3@v3.0.0-20240315174635-e72a50fcb64e/agent_handlers.go:64 +0x37
```

**Expected behavior**

The agent should continue as normal.

**Are you using NetBird Cloud?**

NetBird Cloud

**NetBird version**

v0.35.2

**NetBird status -dA output:**

If applicable, add the `netbird status -dA' command output.

**Do you face any (non-mobile) client issues?**

Please provide the file created by `netbird debug for 1m -AS`.
We advise reviewing the anonymized files for any remaining PII.

**Screenshots**

If applicable, add screenshots to help explain your problem.

**Additional context**

Add any other context about the problem here.

--- a/client/internal/peer/worker_ice.go
+++ b/client/internal/peer/worker_ice.go
@@ -255,6 +255,10 @@ func (w *WorkerICE) closeAgent(cancel context.CancelFunc) {
 	defer w.muxAgent.Unlock()
 
 	cancel()
+	if w.agent == nil {
+		return
+	}
+
 	if err := w.agent.Close(); err != nil {
 		w.log.Warnf("failed to close ICE agent: %s", err)
 	}

